std/algorithm/comparison.d

```diff
@@ -1241,7 +1248,14 @@ MaxType!T max(T...)(T args)
 
     //Do the "max" proper with a and b
     import std.functional : lessThan;
-    immutable chooseB = lessThan!(T0, T1)(a, b);
+    import std.math : isNaN;
+    static if (propagateNaN && isFloatingPoint!T1)
+        immutable chooseB = lessThan(a, b) || isNaN(b);
+    else static if (!propagateNaN && isFloatingPoint!T0)
```

Comment:
Um.  Why?  I mean, current behaviour is filed as a bug, so I don't think
backwards compatibility issues really arise here.  And frankly, of all
the things to do in presence of NaN, always returning the first argument
seems like one of the least attractive ones.  Unless speed is the issue,
but then the question is whether the standard library should be fast or
correct.